### PR TITLE
1262-publishedat-reindex

### PR DESCRIPTION
**Pull request by man-git007**
*[x] publishhed at update
*[x] published_at updated in master and metadata both
*[x] published_at update route updated

### CODE REVIEW

File: app/Controllers/ApiController.php [hunk 1]
```diff
@@ -40,12 +40,9 @@ public function metadata()
                 'updated_at'           => $this->request->request->get('updated_at'),
                 'total_pages'          => $this->request->request->get('total_pages'),
                 'supporting_contracts' => $this->request->request->get('supporting_contracts'),
+                'published_at'         => $this->request->request->get('published_at'),
             ];
 
-            if (!empty($this->request->request->get('published_at'))) {
```
Comment (bikramtuladhar) [inline app/Controllers/ApiController.php]: check if date is valid instead of empty check

File: app/Controllers/ApiController.php [hunk 2]
```diff
@@ -181,4 +178,28 @@ public function deleteContractText()
...
+    /**
+     * Updates published_at in elastic search index
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function updatePublishedAtIndex()
+    {
+        try {
+            $params           = $this->request->request->all();
+            $recent_contracts = $params['recent_contracts'];
```
Comment (bikramtuladhar) [inline app/Controllers/ApiController.php]: extract code to backupMetadata method

File: app/Services/MetadataService.php [hunk 3]
```diff
@@ -34,13 +33,19 @@ public function __construct()
...
-            $params['id'] = $metaData['id'];
-            $document     = $this->es->exists($params);
-            $metadata     = json_decode($metaData['metadata']);
-            $createdBy    = json_decode($metaData['created_by']);
-            $updatedBy    = json_decode($metaData['updated_by']);
-            $data         = [
+            $params                 = $this->getIndexType();
+            $params['id']           = $metaData['id'];
+            $document               = $this->es->exists($params);
+            $metadata               = json_decode($metaData['metadata']);
```
Comment (bikramtuladhar) [inline app/Services/MetadataService.php]: is it multi-dimension json data ? if it is; then set second flag true for associative array output json_decode($metaData['metadata'],true);

File: app/route.php [hunk 4]
```diff
@@ -7,4 +7,5 @@
 $router->addRoute('POST', '/contract/delete', 'App\Controllers\ApiController::deleteContract');
 $router->addRoute('POST', '/contract/delete/metadata', 'App\Controllers\ApiController::deleteContractMetadata');
 $router->addRoute('POST', '/contract/delete/text', 'App\Controllers\ApiController::deleteContractText');
-$router->addRoute('POST', '/contract/delete/annotation', 'App\Controllers\ApiController::deleteContractAnnotation');
\ No newline at end of file
+$router->addRoute('POST', '/contract/delete/annotation', 'App\Controllers\ApiController::deleteContractAnnotation');
+$router->addRoute('POST', '/contract/published_at/update', 'App\Controllers\ApiController::updatePublishedAtIndex');
```
Comment (bikramtuladhar) [inline app/route.php]: url should not contain underscore use dash instead of underscore

File: app/Controllers/ApiController.php (hunk 1 above)
Comment (man-git007) [reply app/Controllers/ApiController.php]: Done

File: app/Controllers/ApiController.php (hunk 2 above)
Comment (man-git007) [reply app/Controllers/ApiController.php]: Will be used only once

File: app/Services/MetadataService.php (hunk 3 above)
Comment (man-git007) [reply app/Services/MetadataService.php]: It will give us object instead of array

File: app/route.php (hunk 4 above)
Comment (man-git007) [reply app/route.php]: It will be used only once

**Pull request merged**